app/src/main/java/com/example/concordia_campus_guide/Adapters/RoutesAdapter.java

```diff
@@ -0,0 +1,208 @@
+package com.example.concordia_campus_guide.Adapters;
+import android.content.Context;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import com.example.concordia_campus_guide.ClassConstants;
+import com.example.concordia_campus_guide.Models.Routes.Bus;
+import com.example.concordia_campus_guide.Models.Routes.Car;
+import com.example.concordia_campus_guide.Models.Direction;
+import com.example.concordia_campus_guide.Models.Routes.Route;
+import com.example.concordia_campus_guide.Models.Routes.Subway;
+import com.example.concordia_campus_guide.Models.Routes.Train;
+import com.example.concordia_campus_guide.Models.Routes.TransportType;
+import com.example.concordia_campus_guide.Models.Routes.Walk;
+import com.example.concordia_campus_guide.R;
+import com.google.android.flexbox.FlexboxLayout;
+
+import java.util.List;
+import androidx.annotation.NonNull;
+
+import org.w3c.dom.Text;
+
+public class RoutesAdapter extends ArrayAdapter<Route> {
+
+    List<Route> routes;
+    Context context;
+
+    // UI elements
+    ImageView mainTransportType;
+    FlexboxLayout details;
+    TextView duration;
+    LinearLayout bottom;
+    TextView arrivalAndDepartureTime;
+    LinearLayout routeOverviewLayout;
+    LinearLayout main;
+
+    public RoutesAdapter(@NonNull Context context, int textViewResourceId, @NonNull List<Route> routes) {
+        super(context, textViewResourceId, routes);
+        this.routes = routes;
+        this.context = context;
+    }
+
+    @Override
+    public View getView(int position, View convertView, ViewGroup parent) {
+        convertView = LayoutInflater.from(this.getContext()).inflate(R.layout.list_routes, parent, false);
+
+        routeOverviewLayout = convertView.findViewById(R.id.routeOverviewLayout);
+        main = convertView.findViewById(R.id.main);
+        mainTransportType = convertView.findViewById(R.id.mainTransportType);
+        details = convertView.findViewById(R.id.details);
```

Comment:
same here I think it is always good to have small methods, you could break it into a initUiComponents()